File: app/src/main/java/com/buel/holyhelper/utils/SortMapUtil.java
package com.buel.holyhelper.utils;

import com.buel.holyhelper.data.CommonData;
import com.buel.holyhelper.model.HolyModel;
import com.orhanobut.logger.LoggerHelper;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.Iterator;
import java.util.List;
import java.util.Map;
import java.util.TreeMap;

/**
 * Created by blue7 on 2018-06-20.
 */

public class SortMapUtil {

    public static List getSortGroupList(Map<String, HolyModel.groupModel> map) {

        HashMap<String, HolyModel.groupModel> tempMap = new HashMap<>();

        for (Map.Entry<String, HolyModel.groupModel> elem : map.entrySet()) {
            HolyModel.groupModel elemTemp = elem.getValue();
            elemTemp.uid = elem.getKey();
            String tempName = elemTemp.name;
            tempMap.put(tempName, elemTemp);
        }

        List<HolyModel.groupModel> tempList = new ArrayList<>();

        TreeMap<String, HolyModel.groupModel> tm = new TreeMap<>(tempMap);
        Iterator<String> iteratorKey = tm.keySet().iterator();   //키값 오름차순 정렬(기본)
        //Iterator<String> iteratorKey = tm.descendingKeySet().iterator(); //키값 내림차순 정렬

        while (iteratorKey.hasNext()) {
            String key = iteratorKey.next();
            tempList.add(tm.get(key));
        }
        return tempList;
    }

    public static List getSortTeamList() {
        LoggerHelper.d("getSortTeamList 1 ");
        HashMap<String, HolyModel.groupModel.teamModel> teamMap
                = (HashMap<String, HolyModel.groupModel.teamModel>) CommonData.getGroupModel().team;


        LoggerHelper.d("getSortTeamList CommonData.getGroupModel().teamModel : " + CommonData.getGroupModel().team);

        HashMap<Integer, HolyModel.groupModel.teamModel> tempMap = new HashMap<>();

        for (Map.Entry<String, HolyModel.groupModel.teamModel> elem : teamMap.entrySet()) {
            HolyModel.groupModel.teamModel elemTemp = elem.getValue();
            LoggerHelper.d("getSortTeamList elem.getKey() : " + elem.getKey());
            elemTemp.uid = elem.getKey();
            int tempCnt = getInteger(elemTemp.name);
            tempMap.put(tempCnt, elemTemp);
        }

        LoggerHelper.d("getSortTeamList tempMap :  " + tempMap);

        List<HolyModel.groupModel.teamModel> tempList = new ArrayList<>();

        TreeMap<Integer, HolyModel.groupModel.teamModel> tm = new TreeMap<>(tempMap);
        Iterator<Integer> iteratorKey = tm.keySet().iterator();   //키값 오름차순 정렬(기본)
        //Iterator<String> iteratorKey = tm.descendingKeySet().iterator(); //키값 내림차순 정렬

        while (iteratorKey.hasNext()) {
            Integer key = iteratorKey.next();
            tempList.add(tm.get(key));
        }

        LoggerHelper.d("getSortTeamList tempList :  " + tempList);

        return tempList;
    }

    public static List getSortMemberMap(Map<String, HolyModel.memberModel> map) {

        HashMap<String, HolyModel.memberModel> tempMap = new HashMap<>();

        for (Map.Entry<String, HolyModel.memberModel> elem : map.entrySet()) {
            HolyModel.memberModel elemTemp = elem.getValue();
            elemTemp.uid = elem.getKey();

            tempMap.put(elemTemp.name, elemTemp);
        }

        List<HolyModel.memberModel> tempList = new ArrayList<>();

        TreeMap<String, HolyModel.memberModel> tm = new TreeMap<>(tempMap);
        Iterator<String> iteratorKey = tm.keySet().iterator();   //키값 오름차순 정렬(기본)
        //Iterator<String> iteratorKey = tm.descendingKeySet().iterator(); //키값 내림차순 정렬

        while (iteratorKey.hasNext()) {
            String key = iteratorKey.next();
            tempList.add(tm.get(key));
        }
        return tempList;
    }

    public static List getSortHolyList(ArrayList<HolyModel> holyModels) {

        HashMap<Integer, HolyModel> tempMap = new HashMap<>();

        Integer c = 0;
        for (HolyModel elem : holyModels) {
            HolyModel elemTemp = elem;
            tempMap.put(c, elemTemp);
            c++;
        }

        List<HolyModel> tempList = new ArrayList<>();

        TreeMap<Integer, HolyModel> tm = new TreeMap<>(tempMap);
        Iterator<Integer> iteratorKey = tm.keySet().iterator();   //키값 오름차순 정렬(기본)
        //Iterator<String> iteratorKey = tm.descendingKeySet().iterator(); //키값 내림차순 정렬

        while (iteratorKey.hasNext()) {
            Integer key = iteratorKey.next();
            tempList.add(tm.get(key));
        }
        return tempList;
    }

    public static List getSortMemberList(ArrayList<HolyModel.memberModel> members) {

        HashMap<String, HolyModel.memberModel> tempMap = new HashMap<>();

        for (HolyModel.memberModel elem : members) {
            HolyModel.memberModel elemTemp = elem;
            tempMap.put(elemTemp.name + "|" + elemTemp.uid, elemTemp);
        }

        List<HolyModel.memberModel> tempList = new ArrayList<>();

        TreeMap<String, HolyModel.memberModel> tm = new TreeMap<>(tempMap);
        Iterator<String> iteratorKey = tm.keySet().iterator();   //키값 오름차순 정렬(기본)
        //Iterator<String> iteratorKey = tm.descendingKeySet().iterator(); //키값 내림차순 정렬

        while (iteratorKey.hasNext()) {
            String key = iteratorKey.next();
            tempList.add(tm.get(key));
        }
        return tempList;
    }

    /**
     * 문자열의 숫자를 뽑아 리턴
     *
     * @param str 숫자가 뒤에 들어가도록...
     * @return 리턴되는 숫자
     */
    public static Integer getInteger(String str) {
        String tempStr = "";

        try {
            int tempNum = Integer.parseInt(str);
            if (tempNum >= 0) {
                return tempNum;
            }
        }catch (Exception e){
            LoggerHelper.d("getInteger  : [ " + str + " ] 은 문자열입니다.");
        }

        //charAt를 이용하여 숫자가 아니면 넘기는 식으로 해서 뽑아 낼 수 있다.
        for (int i = 0; i < str.length(); i++) {
            // 48 ~ 57은 아스키 코드로 0~9이다.
            if (48 <= str.charAt(i) && str.charAt(i) <= 57)
                tempStr += str.charAt(i);
        }
        return Integer.valueOf(tempStr);
    }
}

File: app/src/main/java/com/buel/holyhelpers/utils/SortMapUtil.kt
package com.buel.holyhelpers.utils

import com.buel.holyhelpers.data.CommonData
import com.buel.holyhelpers.model.HolyModel
import com.buel.holyhelpers.model.HolyModel.groupModel
import com.buel.holyhelpers.model.HolyModel.groupModel.teamModel
import com.buel.holyhelpers.model.HolyModel.memberModel
import com.orhanobut.logger.LoggerHelper
import java.util.*

/**
 * Created by blue7 on 2018-06-20.
 */
object SortMapUtil {
    @JvmStatic
    fun getSortGroupList(map: Map<String, groupModel>): List<groupModel> {
        var tempList:MutableList<groupModel>  = map.values as MutableList<groupModel>
        tempList.sortBy { it.name }
        LoggerHelper.d("getSortTeamList tempList :  $tempList")
        return tempList
    }

    //키값 오름차순 정렬(기본)
    //Iterator<String> iteratorKey = tm.descendingKeySet().iterator(); //키값 내림차순 정렬
    @JvmStatic
    val sortTeamList: List<*>
        get() {
            LoggerHelper.d("getSortTeamList 1 ")
            var teamMap = CommonData.teamMap as HashMap<String, teamModel>
            var tempList:MutableList<teamModel>  = teamMap.values as MutableList<teamModel>
            tempList.sortBy { it.name }
            LoggerHelper.d("getSortTeamList tempList :  $tempList")
            return tempList
        }

    fun getSortMemberMap(map: Map<String?, memberModel>): List<*> {
        val tempMap = HashMap<String, memberModel>()
        for ((key, elemTemp) in map) {
            elemTemp.uid = key
            tempMap[elemTemp.name] = elemTemp
        }
        val tempList: MutableList<memberModel?> = ArrayList()
        val tm = TreeMap(tempMap)
        val iteratorKey: Iterator<String> = tm.keys.iterator() //키값 오름차순 정렬(기본)
        //Iterator<String> iteratorKey = tm.descendingKeySet().iterator(); //키값 내림차순 정렬
        while (iteratorKey.hasNext()) {
            val key = iteratorKey.next()
            tempList.add(tm[key])
        }
        return tempList
    }

    fun getSortHolyList(holyModels: ArrayList<HolyModel>): List<*> {
        val tempMap = HashMap<Int, HolyModel>()
        var c = 0
        for (elem in holyModels) {
            tempMap[c] = elem
            c++
        }
        val tempList: MutableList<HolyModel?> = ArrayList()
        val tm = TreeMap(tempMap)
        val iteratorKey: Iterator<Int> = tm.keys.iterator() //키값 오름차순 정렬(기본)
        //Iterator<String> iteratorKey = tm.descendingKeySet().iterator(); //키값 내림차순 정렬
        while (iteratorKey.hasNext()) {
            val key = iteratorKey.next()
            tempList.add(tm[key])
        }
        return tempList
    }

    fun getSortMemberList(members: ArrayList<memberModel>): List<*> {
        val tempMap = HashMap<String, memberModel>()
        for (elem in members) {
            tempMap[elem.name + "|" + elem.uid] = elem
        }
        val tempList: MutableList<memberModel?> = ArrayList()
        val tm = TreeMap(tempMap)
        val iteratorKey: Iterator<String> = tm.keys.iterator() //키값 오름차순 정렬(기본)
        //Iterator<String> iteratorKey = tm.descendingKeySet().iterator(); //키값 내림차순 정렬
        while (iteratorKey.hasNext()) {
            val key = iteratorKey.next()
            tempList.add(tm[key])
        }
        return tempList
    }

    /**
     * 문자열의 숫자를 뽑아 리턴
     *
     * @param str 숫자가 뒤에 들어가도록...
     * @return 리턴되는 숫자
     */
    @JvmStatic
    fun getInteger(str: String?): Int {

        if(str.isNullOrBlank())return -1
        var tempStr = ""
        try {
            val tempNum = str.toInt()
            if (tempNum >= 0) {
                return tempNum
            }
        } catch (e: Exception) {
            LoggerHelper.d("getInteger  : [ $str ] 은 문자열입니다.")
        }
        //charAt를 이용하여 숫자가 아니면 넘기는 식으로 해서 뽑아 낼 수 있다.
        for (i in 0 until str.length) { // 48 ~ 57은 아스키 코드로 0~9이다.
            if (48 <= str[i].toInt() && str[i].toInt() <= 57) tempStr += str[i]
        }
        return if (str.length == 0) {
            0
        } else Integer.valueOf(tempStr)
    }
}